delfin/drivers/dell_emc/scaleio/rest_handler.py

```diff
@@ -0,0 +1,397 @@
+                }
+                list_volumes.append(volume_map)
+            return list_volumes
+        except exception.DelfinException as err:
+            err_msg = "Get Storage volume error: %s" % err.msg
+            LOG.error(err_msg)
+            raise err
+        except Exception as e:
+            LOG.error("Get Storage volume error: %s", six.text_type(e))
+            raise exception.InvalidResults(e)
+
+    def list_disks(self, storage_id):
+        disks_list = []
+        try:
+            disks_json = self.get_rest_info(consts.REST_SCALEIO_DISKS)
+            for json_disk in (disks_json or []):
+                device_status = json_disk.get('deviceState')
+                capacity = json_disk.get('maxCapacityInKb')
+                status = constants.DiskStatus.NORMAL
+                if device_status != 'Normal':
+                    status = constants.DiskStatus.OFFLINE
+                data_map = {
+                    'native_disk_id': json_disk.get('id'),
+                    'name': json_disk.get('name'),
+                    'status': status,
+                    'storage_id': storage_id,
+                    'native_disk_group_id': json_disk.get('sdsId'),
+                    'serial_number': json_disk.get('id'),
+                    'capacity': int(capacity) * units.Ki,
+                    'health_score': status
+                }
+                disks_list.append(data_map)
+            return disks_list
+        except exception.DelfinException as err:
+            err_msg = "Get Storage disk error: %s" % err.msg
+            LOG.error(err_msg)
+            raise err
+        except Exception as e:
+            LOG.error("Get Storage disk error: %s", six.text_type(e))
+            raise exception.InvalidResults(e)
+
+    def list_alerts(self, query_para=None):
+        alert_list = []
+        try:
+            storage_alert = self.get_rest_info(consts.REST_SCALEIO_ALERT)
+            for json_alert in (storage_alert or []):
+                match_key = json_alert.get('id') + json_alert.get('name')
+                occur_time = json_alert.get('startTime')
+                datetime_obj = datetime.datetime.strptime(
+                    occur_time, consts.DATETIME_UTC_FORMAT)
+                alert_time = int(time.mktime(datetime_obj.timetuple()) *
+                                 consts.DEFAULT_ALERTS_TIME_CONVERSION
+                                 + datetime_obj.microsecond /
+                                 consts.DEFAULT_ALERTS_TIME_CONVERSION)
+                if not alert_util.is_alert_in_time_range(query_para,
+                                                         alert_time):
+                    continue
+                alert_type = json_alert.get('alertType')
+                alert_model = {
+                    'alert_id': json_alert.get('id'),
+                    'alert_name': alert_type + json_alert.get('name'),
+                    'severity': json_alert.get('severity'),
```

Comment:
Can the alarm severity obtained from the device be direectly adapted to the alarm servity defined in the code? if not, pls handle the alarm.

Reply:
Added data for determining alarm levels.
![image](https://user-images.githubusercontent.com/88076879/172275932-e181487a-655d-49ef-94ec-8e0d6d0b24c3.png)